provider/aws/aws_test.go

```diff
@@ -1603,11 +1606,227 @@ func TestAWSBatchChangeSetExceedingNameChange(t *testing.T) {
 		)
 	}
 
-	batchCs := batchChangeSet(cs, testLimit)
+	batchCs := batchChangeSet(cs, testLimit, defaultBatchChangeSizeBytes, defaultBatchChangeSizeValues)
 
 	require.Equal(t, 0, len(batchCs))
 }
 
+func TestAWSBatchChangeSetExceedingBytesLimit(t *testing.T) {
+	var cs Route53Changes
+	const testCount = 50
+	const testLimit = 100
+	const groupSize = 2
+	// Bytes for each name
+	var testBytes = len([]byte("1.2.3.4")) + len([]byte("test-record"))
+	// testCount / groupSize / (testLimit // bytes)
+	var expectedBatchCountFloat = float64(testCount) / float64(groupSize) / float64(testLimit/testBytes)
+	// Round up
+	var expectedBatchCount = int(math.Floor(expectedBatchCountFloat + 0.5))
```

Comment:
expectedBatchCount = floor((50 / 2 / (100/18)) + 0.5) = floor(25 / 5.55 + 0.5) = floor(4.5 + 0.5) = floor(5.0) = 5
Why not `math.Ceil(expectedBatchCountFloat)` instead of having the arbitrary `+ 0.5` ?
Or don't calculate and set expected numbers hard coded.

Reply:
same for all other tests

Reply:
@Megum1n I don't mind having a compute but the `+ 0.5` is so "arbitrary". I hope you understand what I mean. :)